pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -1148,18 +1146,32 @@ func buildDefaultCluster(env *model.Environment, name string, discoveryType apiv
 		}
 	}
 
-	defaultTrafficPolicy := buildDefaultTrafficPolicy(env, discoveryType)
+	defaultTrafficPolicy := buildDefaultTrafficPolicy(env, discoveryType, direction, port)
 	applyTrafficPolicy(env, cluster, defaultTrafficPolicy, nil, nil, "",
 		DefaultClusterMode, direction)
 	return cluster
 }
 
-func buildDefaultTrafficPolicy(env *model.Environment, discoveryType apiv2.Cluster_DiscoveryType) *networking.TrafficPolicy {
+func buildDefaultTrafficPolicy(env *model.Environment, discoveryType apiv2.Cluster_DiscoveryType,
+	direction model.TrafficDirection, port *model.Port) *networking.TrafficPolicy {
 	lbPolicy := DefaultLbType
 	if discoveryType == apiv2.Cluster_ORIGINAL_DST {
 		lbPolicy = networking.LoadBalancerSettings_PASSTHROUGH
 	}
+
+	// Set default circuit breaking for Envoy graceful shutdown
+	var outlierDetectionPolicy *networking.OutlierDetection
+	if direction == model.TrafficDirectionOutbound && port.Protocol.IsHTTP() {
```

Comment:
I think my logic is incorrect here. What I need is the upstream listener has an HTTP connection manager, gRPC matches `IsHTTP()` but would not have a connection manager. Could someone confirm that the only protocols that would have this are HTTP and HTTP2?